pyproject.toml

```diff
@@ -10,129 +25,202 @@ authors = [{ name = "Catalyst Cooperative", email = "pudl@catalyst.coop" }]
+channels = ["conda-forge"]
+
+[tool.pixi.dependencies]
+# Runtime dependencies
+alembic = ">=1.17"
+boto3 = ">=1.35"
+bottleneck = ">=1.5.0"
+"catalystcoop.dbfread" = "<3.1,>=3.0"
+"catalystcoop.ferc_xbrl_extractor" = "<2,>=1.7.3"
+# I/O bug in click 8.3.[0-1] https://github.com/pallets/click/issues/3110
+click = "<8.2 | >8.3.1"
+coloredlogs = ">=14.0"
+coverage = ">=7.10"
+curl = ">=8.17"
+dagster = ">=1.12.7"
+dagster-dbt = ">=1!0.28.7"
+dagster-dg-cli = ">=1.12.7"
+dagster-postgres = ">=1!0.28.7"
+dagster-webserver = ">=1.12.7"
+dbt-core = ">=1.10.13"
+dbt-duckdb = ">=1.9.4"
+deepdiff = ">=8"
+doc8 = ">=1.1"
+email-validator = ">=1.0.3"
+frictionless = "<6,>=5"
+fsspec = ">=2025"
+furo = ">=2024"
+gcsfs = ">=2025"
+gdal = "==3.12.1"
+geoarrow-pyarrow = ">=0.2"
+geopandas = ">=1.1"
+google-cloud-sdk = ">=535"
+grpcio = "==1.73.1"
+grpcio-health-checking = "==1.73.1"
+grpcio-status = "==1.73.1"
+hypothesis = ">=6.123"
+jellyfish = ">=1"
+jinja2 = ">=3.1"
+jupyter = "*"
+jupyter-lsp = "*"
+jupyter-resource-usage = "*"
+jupyterlab = ">4.4"
+jupyterlab-lsp = "*"
+matplotlib-base = ">=3.10"
+memray = ">=1.18.0"
+mlflow = ">=3.8"
+nbconvert = ">=7"
+nbformat = ">=5.10"
+networkx = ">=3.6"
+nodejs = ">=24"
+numba = ">=0.63.1"
+numexpr = ">=2.10"
+numpy = ">=2.3"
+packaging = ">=25"
+pandas = ">=2.3"
+pandoc = ">=3.7"
+pandera-core = ">=0.27"
+parallel = ">=20250822"
+pip = ">=25"
+polars = ">=1.36"
+pre-commit = ">=4"
+prettier = ">=3.6,<4"
+pyarrow = "<21,>=20"
+pydantic = ">=2.12"
+pydantic-settings = ">=2.11"
+pygraphviz = "*"
+pytest = ">=9"
+pytest-console-scripts = ">=1.4"
+pytest-cov = ">=7"
+pytest-mock = ">=3.15"
+pytest-order = ">=1.3"
+pytest-xdist = ">=3.8"
+python = ">=3.13,<3.14.0a0"
+python-build = ">=1.2"
+python-calamine = ">=0.6.1"
+python-dotenv = ">=1"
+python-duckdb = ">=1.4"
+pytz = ">=2025"
+pyyaml = ">=6"
+readthedocs-sphinx-ext = ">=2"
+requests = ">=2.32"
+responses = ">=0.25"
+ruff = ">=0.14"
+ruff-lsp = "*"
+s3fs = ">=2025"
+scikit-learn = ">=1.8"
+scipy = ">=1.16"
+shapely = ">=2.1.1"
+sphinx = ">=8"
+sphinx-autoapi = ">=3.6.1"
+sphinx-issues = ">=5"
+sphinx-reredirects = ">=1"
+sphinx-tabs = ">=3.4"
+sphinxcontrib-bibtex = ">=2.6"
+sphinxcontrib-googleanalytics = ">=0.4"
+splink = ">=4.0.10"
+sqlalchemy = ">=2"
+sqlite = ">=3.51"
+terraform = ">=1.14"
+timezonefinder = ">=8"
+typos = ">=1.40"
+universal-pathlib = ">=0.3.7"
+urllib3 = ">=2.4"
+xlsxwriter = ">=3.2"
+zip = ">=3.0"
+
+[tool.pixi.pypi-dependencies]
+# Install only the local package in editable mode with no dependencies.
+# All dependencies are explicitly managed via conda-forge in [tool.pixi.dependencies] above.
+"catalystcoop.pudl" = { path = ".", editable = true, extras = [] }
+
+[tool.pixi.activation.env]
+# Reusable variables for pixi tasks
+ETL_FAST_YML = "src/pudl/package_data/settings/etl_fast.yml"
+ETL_FULL_YML = "src/pudl/package_data/settings/etl_full.yml"
+PYTEST_ARGS = "--durations 20 --cov-fail-under=0"
+COVARGS = "--append"
+PIXI_LOCKED = "true"
```

Comment:
Setting `PIXI_LOCKED` means that by default if you run `pixi install` it will just install the dependencies found in `pixi.lock` rather than re-solving the environment and updating the dependencies.

Resolving and updating is very fast, so it's not a hassle, but this is intended to try and limit dependency churn, and try to keep the big updates limited to the weekly `update-lockfiles` workflow.

If you explicitly run `pixi update` or `pixi add` it will still update the lockfile.

If there's an inconsistency between `pyproject.toml` and `pixi.lock` then `pixi install` will fail (and it runs every time you do any `pixi run` command)